xapian-applications/omega/date.cc

```diff
@@ -144,7 +147,19 @@ date_range_filter(const string & date_start, const string & date_end,
 {
     int y1, m1, d1, y2, m2, d2;
     if (!date_span.empty()) {
-	time_t secs = atoi(date_span.c_str()) * (24 * 60 * 60);
+	errno = 0;
+	long long str_to_longlong = strtoll(date_span.c_str(), NULL, 0);
+	if (errno == ERANGE) {
+	    switch (str_to_longlong) {
+		case LLONG_MIN:
+		    throw (string("Date span underflow:"));
+		case LLONG_MAX:
+		    throw (string("Date span overflow:"));
+	    }
```

Comment:
If somebody asks for all documents which are dated in a range of length `LLONG_MAX + 1` days then treating the range as `LLONG_MAX` days will return all such documents, as we can't represent dates that far away anyway.  So the clamping which `strtoll()` does is actually fine for our purposes.  The same is true for `LLONG_MIN`.